src/odb/src/db/dbTechLayer.cpp

```diff
@@ -915,14 +915,14 @@ uint _dbTechLayer::getV55ColIdx(const int& colVal) const
 }
 uint _dbTechLayer::getTwIdx(const int width, const int prl) const
 {
-  int sz = _two_widths_sp_idx.size();
-  for (int i = 0; i < sz; i++) {
-    if (width <= _two_widths_sp_idx[i])
-      return std::max(0, i - 1);
-    if (_two_widths_sp_prl[i] != -1 && prl <= _two_widths_sp_prl[i])
-      return std::max(0, i - 1);
+  auto pos = --(std::lower_bound(_two_widths_sp_idx.begin(), _two_widths_sp_idx.end(), width));
```

Comment:
If pos is begin() then -- is an invalid operation.  It would be better to do 

if (pos == _two_widths_sp_idx.begin()) {... } else { --pos; .... }

Reply:
That's why I use max of 0 and the index if it's -1 in the next line. But I guess it makes more sense to prevent it from the beginning.

Reply:
--begin() is undefined behavior and isn't guaranteed to be -1.  It will likely work in many implementations but there is no guarantee.